### PR TITLE
Feature: Sign out/Avatar in Navigation

### PR DESCRIPTION
**Pull request by camitaylor**
# [Sign out/Avatar in Navigation](https://trello.com/c/dxMOmEra/32-sign-out-avatar-in-navigation)

## A short description of what this PR does.

As a user, I want to be able to see my avatar in the navigation menu and be able to sign out through a dropdown.

## UI Images
Screen Shot 2021-09-27 at 2 46 09 PM
---------------------------------------------------------------------------------------------------------------------

Screen Shot 2021-09-27 at 2 46 20 PM
---------------------------------------------------------------------------------------------------------------------

Screen Shot 2021-09-27 at 2 46 55 PM
---------------------------------------------------------------------------------------------------------------------

Screen Shot 2021-09-27 at 2 47 08 PM

## How was it tested?
This was linted with Yarn and tested with Jest.

Screen Shot 2021-09-08 at 1 24 40 PM
Screen Shot 2021-09-08 at 1 25 04 PM
Screen Shot 2021-09-08 at 1 25 22 PM

### Yarn Lint
Screen Shot 2021-09-08 at 9 41 00 AM

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/aaronshivers/snapit-hacktoberfest/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

### CODE REVIEW

**Review by aaronshivers — CHANGES_REQUESTED** [review]
💡 Can you add screenshots of the ui?

File: src/components/AvatarMenu/AvatarMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          size="small"
+          alt="User's Name"
+          src="/static/images/avatar/1.jpg"
+          button="true"
+          onClick={handleClickListItem}
+          data-test="avatar-button"
+        />
+        <h3 data-test="username-display">Username</h3>
+        <div>
+          {/* className={classes.root}> */}
```
[the remark this answers is not in the document]
Comment (camitaylor) [reply src/components/AvatarMenu/AvatarMenu.jsx]: Addressed: commit 7d7872b6d7230688815e7aa72d44df109539dcad

File: src/components/AvatarMenu/AvatarMenu.test.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  test("renders without error", () => {
+    const wrapper = setup();
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  test("renders avatar display", () => {
+    const wrapper = setup();
+    const avatarDisplay = findByTestAttribute(wrapper, "avatar-display");
+    expect(avatarDisplay.length).toBe(1);
```
[the remark this answers is not in the document]
Comment (camitaylor) [reply src/components/AvatarMenu/AvatarMenu.test.jsx]: Addressed: commit 7d7872b6d7230688815e7aa72d44df109539dcad

File: src/components/AvatarMenu/AvatarMenu.test.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  });
+
+  test("renders username display", () => {
+    const wrapper = setup();
+    const usernameDisplay = findByTestAttribute(wrapper, "username-display");
+    expect(usernameDisplay.length).toBe(1);
```
[the remark this answers is not in the document]
Comment (camitaylor) [reply src/components/AvatarMenu/AvatarMenu.test.jsx]: Addressed: commit 7d7872b6d7230688815e7aa72d44df109539dcad

File: src/components/AvatarMenu/AvatarMenu.test.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  });
+
+  test("clicking button displays avatar menu", () => {
+    const wrapper = setup();
+    const button = findByTestAttribute(wrapper, "avatar-button");
+    button.simulate("click", new Event("click"));
+    const menuDisplay = findByTestAttribute(wrapper, "menu-display");
+    expect(menuDisplay.length).toBe(1);
```
[the remark this answers is not in the document]
Comment (camitaylor) [reply src/components/AvatarMenu/AvatarMenu.test.jsx]: Addressed: commit 7d7872b6d7230688815e7aa72d44df109539dcad

**Comment by aaronshivers** [comment]
📷 Add screenshots of the UI.

**Comment by aaronshivers** [comment]
🔧 rebase

File: src/components/AvatarMenu/AvatarMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      history.push("/");
+    });
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <div>
```
Comment (aaronshivers) [inline src/components/AvatarMenu/AvatarMenu.jsx:36]: ⛏️ It looks like there's an extra set of `div` tags here.

**Review by aaronshivers — APPROVED** [review]
👍 Approved, but please address and resolve the remaining comment.

File: src/components/AvatarMenu/AvatarMenu.jsx (hunk 5 above)
Comment (camitaylor) [reply src/components/AvatarMenu/AvatarMenu.jsx]: Addressed: commit 1c3755dbcd379ff9772a77a70f14a4cba198d8d1

**Comment by aaronshivers** [comment]
🔧 rebase

**Review by aaronshivers — APPROVED** [review]

**Pull request merged**